lib/vanagon/platform.rb

```diff
@@ -469,13 +469,27 @@ def generate_compiled_archive(project)
       final_archive = "output/#{name_and_version_and_platform}.tar.gz"
       archive_directory = "#{project.name}-archive"
       metadata = project.build_manifest_json(true)
+
+      # previously, we weren't properly handling the case of custom BOM paths.
+      # If we have a custom BOM path, during Makefile execution, the top-level
+      # BOM is moved to the custom path. So, when cleaning up BOMs for non-custom
+      # paths we just want to remove the BOM at the top level of the tarball.
+      # But, if we have a custom BOM path we want to move it back to where it
+      # was prior to the Makefile execution so we can preserve it as an artifact
+      # but not leave it to conflict if it's installed in the same custom path
+      # as a project using this archive.
+      bill_of_materials_command = 'rm -f bill-of-materials'
+      if project.bill_of_materials
+        bill_of_materials_command = "mv .#{project.bill_of_materials.path}/bill-of-materials ../.."
+      end
```

Comment:
![twocents_1914](https://user-images.githubusercontent.com/344926/38295311-70d7c4ac-37a2-11e8-8cc7-e272a18b7c0a.jpg)


Consider this as a more idiomatic Ruby variable assignment:

```ruby
bill_of_materials_command = if project.bill_of_materials
                              "mv .#{project.bill_of_materials.path}/bill-of-materials ../.."
                            else
                              'rm -f bill-of-materials'
                            end
```

(though I think Rubocop might want it aligned under `bill_of_materials_command`, not under `if`)

Reply:
`bill_of_materials_command = project.bill_of_materials ? "mv .#{project.bill_of_materials.path}/bill-of-materials ../.." : 'rm -f bill-of-materials'` seems more idiomatic to me?
but also, I'm totally fine with how it is rn.